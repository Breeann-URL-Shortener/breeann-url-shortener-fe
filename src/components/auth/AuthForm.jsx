import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useAuthError } from '../../state/AuthContext';
import buglogo from '../../assets/minuscurl_logo_bug.png';
import './AuthForm.scss';
import { Link } from 'react-router-dom';

const AuthForm = ({ title, authFn, action, link, redirectMessage }) => {
  const error = useAuthError();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleSubmit = event => {
    event.preventDefault();
    authFn(email, password);
  };

  return (
    <div className="auth-form">
      <section>
        <img src={buglogo} alt="miniscurl bug logo"/>
        <h1>{title}</h1>
      
        {error && <p>{error.message}</p>}
        <form onSubmit={handleSubmit}>
          <label>
            <input
              required
              type="email"
              value={email}
              placeholder="Email"
              onChange={({ target }) => setEmail(target.value)} />
          </label>
          <input
            required
            type="password"
            value={password}
            placeholder="Password"
            onChange={({ target }) => setPassword(target.value)} />

          <button>{action}</button>

          <Link to={link}>
            {redirectMessage}
          </Link>
        </form>
      </section>
    </div>
  );
};

AuthForm.propTypes = {
  title: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  redirectMessage: PropTypes.string.isRequired,
  action: PropTypes.string.isRequired,
  authFn: PropTypes.func.isRequired
};

export default AuthForm;
