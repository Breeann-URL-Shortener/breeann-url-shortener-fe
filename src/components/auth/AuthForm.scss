@import 'src/main';

.auth-form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    justify-self: center;
    text-align: center;
    align-items: center;
    background: $yellow-orange-gradient;
    border-radius: 15px;
    box-shadow: rgba(50, 50, 93, 0.25) 0px 50px 100px -20px, rgba(0, 0, 0, 0.3) 0px 30px 60px -30px, rgba(10, 37, 64, 0.35) 0px -2px 6px 0px inset;
    margin: auto;
    margin-top: 10%;
    width: 50%;
    height: auto;

    @media (max-width: 850px) {  
        width: 80%;
    }
    
    h1 {
        @include sub-header;
    }

    img {
        width: 80px;
        margin-top: 4rem;
    }

    form {
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-items: center;
    }

    input {
        @include input;

        @media (max-width: 850px) {  
            width: auto;
        }


        // &:valid {
        //     // background: url('../../assets/verify.png');
        //     background-color: $secondary-green;
        //     background-size: 20px;
        //     background-repeat: no-repeat;
        //     background-position: 20px 20px;
        // }

    }

    button {
        @include button;
        margin: 2rem;

        @media (max-width: 850px) {  
            width: auto;
        }
    }

    a {
        @include link;
        margin-left: 2rem;
        margin-right: 2rem;
        margin-bottom: 4rem;
    }


}
