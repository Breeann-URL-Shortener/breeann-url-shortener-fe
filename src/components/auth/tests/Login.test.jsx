/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router';
import AuthForm from '../AuthForm';
import { AuthProvider, useLogin } from '../../../state/AuthContext';

describe('DetailsPage component', () => {
  afterEach(() => cleanup());
  const login = jest.fn();
  it('renders DetailsPage', () => {
    const { asFragment } = render(
      <Router>
        <AuthProvider>
          <AuthForm 
            authFn={login}
            title="Welcome Back! Please Login." 
            action="Login" 
            link="/signup"
            redirectMessage="Not yet a member? Click here to sign up."/>
        </AuthProvider>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
