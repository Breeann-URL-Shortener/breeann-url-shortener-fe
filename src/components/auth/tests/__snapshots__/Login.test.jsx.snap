// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage component renders DetailsPage 1`] = `
<DocumentFragment>
  <section
    class="auth-form"
  >
    <img
      alt="miniscurl bug logo"
      src="/path/image"
    />
    <h1>
      Welcome Back! Please Login.
    </h1>
    <form>
      <label>
        <input
          placeholder="Email"
          required=""
          type="email"
          value=""
        />
      </label>
      <input
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <button>
        Login
      </button>
      <a
        href="/signup"
      >
        Not yet a member? Click here to sign up.
      </a>
    </form>
  </section>
</DocumentFragment>
`;
